src/BusinessObject/HealthCheckBo.php

```diff
@@ -0,0 +1,133 @@
+use YapepBase\BusinessObject\BoAbstract;
+use YapepBase\Config;
+use YapepBase\Exception\ConfigException;
+use YapepBase\Exception\ParameterException;
+
+class HealthCheckBo extends BoAbstract
+{
+    protected const BUILT_IN_HEALTH_CHECKS = [
+        'db'      => DbHealthCheck::class,
+        'storage' => StorageHealthCheck::class,
+        'sqs'     => SqsHealthCheck::class,
+    ];
+
+    /**
+     * @throws ConfigException
+     */
+    public function performHealthCheck(bool &$haveErrors = false): array
+    {
+        $config            = Config::getInstance();
+        $healthCheckConfig = $this->getHealthCheckConfig();
+        $healthChecks      = $this->getHealthChecks();
+        $results           = [];
+
+        foreach ($healthCheckConfig as $type => $checks) {
+            $healthCheck = $this->getHealthCheck($healthChecks, $type);
+
+            foreach ($checks as $name => $configOptions) {
+                $this->performSingleHealthCheck($type, $name, $configOptions, $healthCheck, $results);
+            }
+        }
+
+        return $results;
+    }
+
+    /**
+     * @throws ConfigException
+     */
+    protected function getHealthChecks(): array
+    {
+        return array_merge(
+            self::BUILT_IN_HEALTH_CHECKS,
+            Config::getInstance()->get('applicationCommon.healthCheckDefinitions', [])
+        );
+    }
+
+    /**
+     * @throws ConfigException
+     */
+    protected function getHealthCheckConfig(): array
+    {
+        $healthCheckConfig = Config::getInstance()->get('applicationCommon.healthCheck.checks');
+
+        if (!is_array($healthCheckConfig)) {
+            throw new ConfigException('The "applicationCommon.healthCheck.checks" should contain an array');
+        }
+
+        return $healthCheckConfig;
+    }
+
+    /**
+     * @throws ConfigException
+     */
+    protected function getHealthCheck(array $healthChecks, string $type): IHealthCheck
+    {
+        if (!isset($healthChecks[$type])) {
+            throw new ConfigException('Undefined health check type: ' . $type);
+        }
+
+        $healthCheckClass = $healthChecks[$type];
+
+        if (!class_exists($healthChecks[$type])) {
+            throw new ConfigException('Non-existing health check class: ' . $healthCheckClass);
+        }
+
+        $healthCheck = new $healthCheckClass();
+
+        if (!($healthCheck instanceof IHealthCheck)) {
+            throw new ConfigException(
+                'The health check class ' . $healthCheckClass . ' does not implement IHealthCheck'
+            );
+        }
+
+        return $healthCheck;
+    }
+
+    /**
+     * @throws ConfigException
+     */
+    protected function performSingleHealthCheck(
+        string $type,
+        string $name,
+        $configOptions,
+        IHealthCheck $healthCheck,
+        array &$results
+    ): void {
+        $fullName = $type . '.' . $name;
+
+        if (!is_array($configOptions)) {
```

Comment:
Because it's coming from a config, and a hard fail would make it more difficult to track down where/why it failed.

Reply:
Yeah as I suspected, but I still don't like it. It can lead far to "omit" type hints because of more readable errors. If something need further validation or special error messages validate them upfront. 
Actually you already have a method which returns and validates this config. That method (getHealthCheckConfig) should do this validation as well.